source/fswatch.d

```diff
@@ -238,9 +238,11 @@ struct FileWatch
 		import std.string : toStringz, fromStringz;
 		import std.conv : to;
 
-		private int fd, wd;
+		private int fd;
+		private bool[] wds = new bool[1024]; // mark every watch descriptor
 		private ubyte[1024 * 4] eventBuffer; // 4kb buffer for events
 		private pollfd pfd;
+		private string[int] directoryMap; // map every watch descriptor to a directory

```

Comment:
why is this an AA and wds a static array on the GC? I think making a new `private struct FDInfo { int fd; bool watched; string path; }` and having this be a `FDInfo[]` then would be better. Especially when having more than 1024 subdirs.

Reply:
The new code should follow your directions.
I must say that I don't agree for two reasons:
4096 is the standard ulimit value for the number of file descriptors that could be opened, but it could be more than that or less.
The code should dynamically allocate (given that the GC is used in the rest of the library) or call to ulimit and allocate the right amount.
Given that this is not related to my PR, I'll follow your directions.